Penteract/Assets/Scripts/Barrel.cpp

```diff
@@ -29,20 +32,38 @@ void Barrel::Start() {
 		particles = particleAux->GetComponent<ComponentParticleSystem>();
 		audio = particleAux->GetComponent<ComponentAudioSource>();
 	}
+
+	GameObject* particleForTimerAux = GameplaySystems::GetGameObject(particlesForTimerUID);
+	if (particleForTimerAux) {
+		particlesForTimer = particleForTimerAux->GetComponent<ComponentParticleSystem>();
+		audioForTimer = particleForTimerAux->GetComponent<ComponentAudioSource>();
+	}
 }
 
 void Barrel::Update() {
 
+	if (startTimerToDestroy && timerDestroyActivated) {
+		if (particlesForTimer) particlesForTimer->PlayChildParticles();
```

Comment:
You don't really need to call PlayChildParticles everytime. One is enough to start the particles loop

Reply:
It is going to run just one time given that the boolean on line 54 is going to be set as false